### PR TITLE
Jacob/feat/error handlers for ajax calls

### PR DESCRIPTION
**Pull request by Goldenaries658**
Resolving Error Handling, this will need some strong testing. I believe I've dealt with all possible errors but would be good for someone else to run some bad searches

### CODE REVIEW

**Comment by meeday** [comment]
what if the number of albums returned is not four for example search for the artist demon days and you'll see what I mean

File: assets/script.js [hunk 1]
```diff
@@ -104,94 +139,134 @@ function getArtistInfo(userInput) {
...
     callAPI(userInput);
     $("#default-search-input>label").remove();
     $("#album-input>label").toggle();
     $("#artist-input>label").toggle();
   });
 
-  //search on <enter key> pressed
+  // Event listeners
+  // Modal function
+  document.addEventListener("DOMContentLoaded", function () {
```
Comment (meeday) [inline assets/script.js]: this modal function needs to be at the top or the modals don't work for some reason

**Review by meeday — CHANGES_REQUESTED** [review]
move the modal function to the top of the page

File: assets/script.js (hunk 1 above)
Comment (Goldenaries658) [reply assets/script.js]: Outdated, function was scoped within the doc.ready function, now sits at the bottom of the doc. Needs to be with the rest of the event listeners, placed after so it applies last after everything else has been loaded in

**Comment by abc-1211** [comment]
image
Error was found when artist search: hi
image is not loaded

File: assets/script.js (hunk 1 above)
Comment (meeday) [reply assets/script.js]: if thats the case put the function after all of the event listeners right at the bottom

Comment (Goldenaries658) [reply assets/script.js]: That wouldn't follow the basic rules of JS: declare global variables > functions > event listeners.

**Review by meeday — APPROVED** [review]

**Pull request merged**